src/build_vehicle_gui.cpp

```diff
@@ -1010,6 +1304,67 @@ void DisplayVehicleSortDropDown(Window *w, VehicleType vehicle_type, int selecte
 	ShowDropDownMenu(w, _engine_sort_listing[vehicle_type], selected, button, 0, hidden_mask);
 }
 
+bool IsEnginePassFilter(const VehicleFilter& filter, const Engine* engine) {
+	if (!filter.show_hidden_engines && engine->IsHidden(_local_company)) return false;
+	if (!IsEngineBuildable(engine->index, VEH_TRAIN, _local_company)) return false;
+
+	const RailFilter& rail_filter = filter.rail;
+	uint hasPower = engine->GetPower() > 0;
+	bool hasCapacity = engine->CanCarryCargo();
+
+	bool is_locomotive = !hasCapacity && hasPower;
+	bool is_motor_wagon = hasCapacity && hasPower;
+	bool is_wagon = hasCapacity && !hasPower;
+
+	if (!is_locomotive && !is_motor_wagon && !is_wagon) {
+		return false;
```

Comment:
This and the following checks could be turned into single-line.

Reply:
Counter-point: by naming `is_wagon` explicitly, it is more clear what is being checked

Reply:
I'm not advocating renaming anything. I'm advocating turning the multi-line if statements into single-line.

Changing this:
```c++
if (!is_locomotive && !is_motor_wagon && !is_wagon) {
	return false;
}
```
into this:
```c++
if (!is_locomotive && !is_motor_wagon && !is_wagon) return false;
```